### PR TITLE
Support model scoring

### LINKED ISSUE #77: Feature suggestion: goodness metric from model building
Provide a metric (or several) from a model building run. Could be just the auto-encoder avg error or something, or a "overfitting indicator" where the difference between (ratio or similar) the results on training data and some test data is reported, stored, and provided as metadata. Could be a separate test period, or a cross validation / train test type split.

Need to discuss this with the whole team, and then implement. Should also be used when assessing new architectures or hyperparameters.

### LINKED ISSUE #115: When training, do validation and store metric
A n-fold CV, storing and serving the metrics. Probably as a separate step to the actual building using the full training set

### PR DESCRIPTION
**Pull request by milesgranger**
Will close #77 and close #115

### CODE REVIEW

**Review by epa095 — COMMENTED** [review]
I like it. But do we really want to do CV every time, or should it be customizable ?

**Review by epa095 — APPROVED** [review]
I accept because I like it and I trust you to fix the stuff I commented on, and squash the commits to something reasonable

**Review by epa095 — APPROVED** [review]
